crates/subspace-runtime/src/lib.rs

```diff
@@ -846,9 +849,12 @@ fn extract_call_block_object_mapping<I: Iterator<Item = Hash>>(
     }
 }
 
-fn extract_block_object_mapping(block: Block, successful_calls: Vec<Hash>) -> BlockObjectMapping {
+fn extract_block_object_mapping(
+    block: Block,
+    successful_calls: Option<Vec<Hash>>,
```

Comment:
Why option and not an empty vector? I think it is fine to not create object mappings for old clients, we don't put objects on farmnet anyway.